Assets/MixedRealityToolkit.Tests/PlayModeTests/SpatialObserverTests.cs

```diff
@@ -0,0 +1,123 @@
+        [UnityTest]
+        public IEnumerator TestAutoStartObserver()
+        {
+            var mrtkProfile = CreateMRTKTestProfile(TestSpatialAwarenessSysteProfilePath);
+            TestUtilities.InitializeMixedRealityToolkit(mrtkProfile);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+
+            var spatialAwarenesssDataProvider = CoreServices.SpatialAwarenessSystem as IMixedRealityDataProviderAccess;
+            var spatialObserver = spatialAwarenesssDataProvider.GetDataProvider<SpatialObjectMeshObserver.SpatialObjectMeshObserver>();
+            Assert.IsNotNull(spatialObserver, "No SpatialObjectMeshObserver data provider created or found");
+            Assert.IsTrue(spatialObserver.IsRunning);
+
+            CoreServices.SpatialAwarenessSystem.Disable();
+            spatialObserver = spatialAwarenesssDataProvider.GetDataProvider<SpatialObjectMeshObserver.SpatialObjectMeshObserver>();
+            Assert.IsNull(spatialObserver);
+
+            CoreServices.SpatialAwarenessSystem.Enable();
+            spatialObserver = spatialAwarenesssDataProvider.GetDataProvider<SpatialObjectMeshObserver.SpatialObjectMeshObserver>();
+            Assert.IsNotNull(spatialObserver, "No SpatialObjectMeshObserver data provider created or found");
+            Assert.IsTrue(spatialObserver.IsRunning);
+        }
+
+        /// <summary>
+        /// Test case of Manual-Start SpatialObjectMeshObserver observer type and SpatialAwarenessSystem in editor
+        /// Validates that system should not resume observers unless directly told by test
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestManualStartObserver()
+        {
+            var mrtkProfile = CreateMRTKTestProfile(TestSpatialAwarenessSysteProfilePath_ManualStart);
+            TestUtilities.InitializeMixedRealityToolkit(mrtkProfile);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+
+            var spatialAwarenesssDataProvider = CoreServices.SpatialAwarenessSystem as IMixedRealityDataProviderAccess;
+            var spatialObserver = spatialAwarenesssDataProvider.GetDataProvider<SpatialObjectMeshObserver.SpatialObjectMeshObserver>();
+            Assert.IsNotNull(spatialObserver, "No SpatialObjectMeshObserver data provider created or found");
+            Assert.IsFalse(spatialObserver.IsRunning);
+
+            CoreServices.SpatialAwarenessSystem.Disable();
+            spatialObserver = spatialAwarenesssDataProvider.GetDataProvider<SpatialObjectMeshObserver.SpatialObjectMeshObserver>();
+            Assert.IsNull(spatialObserver);
+
+            CoreServices.SpatialAwarenessSystem.Enable();
+            spatialObserver = spatialAwarenesssDataProvider.GetDataProvider<SpatialObjectMeshObserver.SpatialObjectMeshObserver>();
+            Assert.IsNotNull(spatialObserver, "No SpatialObjectMeshObserver data provider created or found");
+            Assert.IsFalse(spatialObserver.IsRunning);
```

Comment:
Any ideas for how to test that the visual of the spatial awareness system is not showing up? To test that turning off the spatial awareness actually hides the mesh. Perhaps you could search for a game object in the scene and test it is disabled.

Reply:
Good idea. It’s actually accessible via the observer interface I believe